Clarify Prometheus TLS settings. (#8962)
istio/istio.io: content/en/docs/ops/integrations/prometheus/index.md

One way to provision Istio certificates for Prometheus is by injecting a sidecar which will rotate SDS certificates and output them to a volume that can be shared with Prometheus.
However, the sidecar should not intercept requests for Prometheus because the Prometheus's model of direct endpoint access is incompatible with Istio's sidecar proxy model.

> To achieve this, configure a cert volume mount at Prometheus server container:

Reviewer: ```suggestion
To achieve this, configure a cert volume mount on the Prometheus server container:
```

{{< text yaml >}}
containers:
  - name: prometheus-server
    ...
    volumeMounts:
      mountPath: /etc/prom-certs/
      name: istio-certs
volumes:
  - emptyDir:
      medium: Memory
    name: istio-certs
{{< /text >}}

> Then add the following annotations to the Prometheus deployment pod template, and deploy it with [sidecar injected](/docs/setup/additional-setup/sidecar-injection/).

Reviewer: ```suggestion
Then add the following annotations to the Prometheus deployment pod template, and deploy it with [sidecar injection](/docs/setup/additional-setup/sidecar-injection/).
```

This configures the sidecar to write a certificate to the shared volume, but without configuring traffic redirection:

{{< text yaml >}}
spec:
  template:
    metadata:
      annotations:
        traffic.sidecar.istio.io/includeInboundPorts: ""   # do not intercept any inbound ports
        traffic.sidecar.istio.io/includeOutboundIPRanges: ""  # do not intercept any outbound traffic
        proxy.istio.io/config: |  # configure an env variable `OUTPUT_CERTS` to write certificates to the given folder
          proxyMetadata:
            OUTPUT_CERTS: /etc/istio-output-certs
        sidecar.istio.io/userVolumeMount: '[{"name": "istio-certs", "mountPath": "/etc/istio-output-certs"}]' # mount the shared volume at sidecar proxy
{{< /text >}}

> Finally, set the scraping job TLS context as follow:

Reviewer: ```suggestion
Finally, set the scraping job TLS context as follows:
```
